### ISSUE TITLE
Auto-Type stops incomplete due to "unsaved DB changes" dialog

### ISSUE THREAD

**Issue by GsakuL** [keepassxreboot/keepassxc#10330]
## Overview
[TIP]:  # ( DO NOT include screenshots of your actual database! )
[NOTE]: # ( Give a BRIEF summary about your problem )
This is more of a "minor annoyance", it's still usable.

## Steps to Reproduce
[NOTE]: # ( Provide a simple set of steps to reproduce this bug. )
1. create/have entry A
2.  save DB
3. create entry B
4. don't save DB
5. use Auto Type for some entry (B in my case because that's the reason I'm adding an entry)

## Expected Behavior
[NOTE]: # ( Tell us what you expected to happen )
a) either check if DB needs saving before starting auto type
b) or ensure auto type is complete(?) before showing save dialog AND bring main window back to front/focus

## Actual Behavior
[NOTE]: # ( Tell us what actually happens )
It fails silently.
Auto Type will fill username, {Tab}, start entering password, but stops midway.
KeePassXC shows the unsaved changes dialog (still in background), because it wants to lock the DB "after" auto type is "done"

## Context
[NOTE]: # ( Give us any additional information you may have. )

[NOTE]: # ( Paste debug info from Help → About here )
KeePassXC - Version 2.7.6
Revision: dd21def

[NOTE]: # ( Pick choices based on your environment )
Operating System: Windows 11
Desktop Env: -
Windowing System: -

**Comment by droidmonkey** [comment]
Auto type will only lock the database if it was locked _before_ starting Auto type and that option is enabled. I'm not sure what scenario would cause your reported behavior.

**Comment by GsakuL** [comment]
i forgot to add: I'm not using global auto type. I even forgot this exists, I used the right click on the entry
![grafik](https://github.com/keepassxreboot/keepassxc/assets/6844160/aba796ef-fda3-4adb-8004-5f88ab85b2a9)
And as far as I can remember, I only changed a few settings (if any at all)

**Comment by droidmonkey** [comment]
Entry level auto type will never lock the database. Do you have your database auto lock timeout set to something really short? 

Application settings -> Security -> timeout lock

Could also happen if you enabled lock on minimize.

**Comment by GsakuL** [comment]
I made a screen recording of my experience, but I just saw your edit.

yes, lock on minimize is enabled. However I like this option in of itself.
![grafik](https://github.com/keepassxreboot/keepassxc/assets/6844160/164075cd-6764-4bde-b1f8-deb5e5e13beb)

for the recording i used a new testing DB, with the same settings. The calculator being open there blocks the unsaved changes dialog box. The dialog would be visible if there is no other window (which might also be a maximized browser window usually instead of calc+notepad)

https://github.com/keepassxreboot/keepassxc/assets/6844160/fafbfd22-b3b1-416f-9911-fd7bf8979454
